portfolio/src/main/webapp/script.js

```diff
@@ -17,12 +17,45 @@
  */
 function addRandomFact() {
   const facts =
-      ['I have a pet dog', 'I like to code'];
+      ['I have a pet dog!', 'I like to code!', 'Marquees are my favourite HTML element!'];
 
   // Pick a random fact.
   const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
-}
\ No newline at end of file
+}
+
+/**
+ * Controls slideshow image display on the page
+ */
+var slideIndices = [0, 0, 0]; // array of slide indices for each slide deck
+for(var i = 0; i < slideIndices.length; i++) {
+    showSlides(i, slideIndices[i]);
+}
+
+function nextSlide(deckNum, direction) {
+  showSlides(deckNum, slideIndices[deckNum] += direction);
+}
+
+// Returns corresponding class name for given deckNum
+function deckNumToName(deckNum) {
+  if(deckNum === 0)
+    return "dog-slides"
+  else if(deckNum == 1)
+    return "place-slides"
+  else
+    return "food-slides"
+}
+
+function showSlides(deckNum, slideIndex) {
+  var slides = document.getElementsByClassName(deckNumToName(deckNum));
```

Comment:
What do you think of only having one img element in the html and setting/changing its src attribute in js instead of having N img tags in the HTML and setting display to none?

Reply:
I could definitely change it to this, I think it would be more readable. I had the images in the HTML before because I didn't have the slideshow.
A tradeoff would be that if the script doesn't load for whatever reason, there would only be one image though.

Reply:
I think that would be more readable. The trade-off would be that if the script doesn't load for any reason there would only be one image but I suppose that doesn't happen very often.

Reply:
I have changed this but I am noticing that the images take longer to load the first time they are viewed and I am unsure how to handle this issue while keeping all the images in the javascript?